node-manager/src/nodemanager.rs

```diff
@@ -565,26 +580,52 @@ impl NodeManager {
     }
 
     #[wasm_bindgen]
-    pub async fn decode_invoice(&self, _invoice: String) -> Result<MutinyInvoice, MutinyJsError> {
-        todo!()
+    pub async fn decode_invoice(&self, invoice: String) -> Result<MutinyInvoice, MutinyJsError> {
+        Invoice::from_str(&invoice)
+            .map(|i| i.into())
+            .map_err(|e| e.into())
```

Comment:
doing a `map` and `map_err` is equivalent to just doing a match, that would be more clean here

Reply:
cleaned up